python/GafferUI/PathListingWidget.py

```diff
@@ -902,20 +909,22 @@ def updateColumnWidths( self ) :
 		self.__recalculatingColumnWidths = True
 
 		header = self.header()
-		numColumnsToResize = header.count() - 1 # leave the last section alone, as it's expandable
+		numColumnsToResize = header.count()
+		if header.stretchLastSection() :
+			numColumnsToResize -= 1 # leave the last section alone, as it's expandable
 
 		if numColumnsToResize != len( self.__columnWidthAdjustments ) :
 			# either the first time in here, or the number of columns has
 			# changed and we want to start again with the offsets.
 			self.__columnWidthAdjustments = [ 0 ] * numColumnsToResize
 
-		del self.__idealColumnWidths[:]
+		self.__idealColumnWidths = [ 0 ] * numColumnsToResize
 		for i in range( 0, numColumnsToResize ) :
+			if header.sectionResizeMode( i ) == QtWidgets.QHeaderView.ResizeMode.Interactive :
+				idealWidth = max( header.sectionSizeHint( i ), self.sizeHintForColumn( i ) )
+				self.__idealColumnWidths[i] = idealWidth
 
-			idealWidth = max( header.sectionSizeHint( i ), self.sizeHintForColumn( i ) )
-			self.__idealColumnWidths.append( idealWidth )
-
-			header.resizeSection( i, idealWidth + self.__columnWidthAdjustments[i] )
+				header.resizeSection( i, idealWidth + self.__columnWidthAdjustments[i] )
```

Comment:
How come we only call `resizeSection` for interactive sections now? I guess it's because Qt is automatically resizing to contents as necessary for the other modes?